tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/EagerOperation.java

```diff
@@ -52,7 +50,8 @@
     this.session = session;
     this.type = type;
     this.name = name;
-    this.nativeRef = new NativeReference(session, this, opNativeHandle, outputNativeHandles);
+    session.attach(this.opHandle = opNativeHandle);
```

Comment:
Can we avoid side effect assignment here? I'd much prefer two lines, one for the assignment and one for the `session.attach` call.